### PR TITLE
Disable hardware acceleration via API that sets pref

### LINKED ISSUE #12887: Disabled hardware acceleration causing startup problems

### PR DESCRIPTION
**Pull request by bbondy**
Fix #12887

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
  - No existing tests around hardware acceleration
- x ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

Test Plan:

Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by bsclifton** [comment]
Does this issue close https://github.com/brave/browser-laptop/issues/5833 ?

File: app/index.js [hunk 1]
```diff
@@ -112,9 +112,6 @@ loadAppStatePromise.then((initialImmutableState) => {
   CrashHerald.init(getSetting(SEND_CRASH_REPORTS, initialSettings))
 
   telemetry.setCheckpointAndReport('state-loaded')
-  if (getSetting(HARDWARE_ACCELERATION_ENABLED, initialSettings) === false) {
```
Comment (bridiver) [inline app/index.js]: I think we still need to handle the case here where the local state pref and appstate pref are out of sync. I'll add an api to check the current value in muon

File: app/index.js [hunk 2]
```diff
@@ -112,8 +112,14 @@ loadAppStatePromise.then((initialImmutableState) => {
...
-    app.disableHardwareAcceleration()
+
+  // Check in case a user is upgrading with this setting.
+  // For non-legacy users, they will always have these settings in sync.
+  if (getSetting(HARDWARE_ACCELERATION_ENABLED, initialSettings) === false &&
+      app.getBooleanPref('hardware_acceleration_mode.enabled')) {
+    // This won't take effect until the next restart of the app, so some users
+    // may have to restart an extra time if UI doesn't work.
```
Comment (bbondy) [inline app/index.js]: Relating to this comment, I was going to relaunch here but ran into an issue. I posted this to investigate it further:
https://github.com/brave/browser-laptop/issues/12896

File: js/stores/appStore.js [hunk 3]
```diff
@@ -122,6 +122,9 @@ const emitChanges = debounce(appStore.emitChanges.bind(appStore), 5)
  */
 function handleChangeSettingAction (state, settingKey, settingValue) {
   switch (settingKey) {
+    case settings.HARDWARE_ACCELERATION_ENABLED:
+      app.setBooleanPref('hardware_acceleration_mode.enabled', settingValue)
```
Comment (bridiver) [inline js/stores/appStore.js]: should probably call this in a setImmediate block

Comment (bbondy) [reply js/stores/appStore.js]: agree but it doesn't change frequently so not going to spend time on it and retest for now.

**Pull request merged**